lib/sidekiq/history/middleware.rb

```diff
@@ -44,18 +44,18 @@ def save_entry_for_worker(worker_status)
           history = "sidekiq:history:#{Time.now.utc.to_date}"
 
           redis.watch(history) do
-          value = redis.get history
+            value = redis.get history
 
-          redis.multi do |multi|
-            if value
-              summary = Sidekiq.load_json(value)[worker].symbolize_keys
-              [:failed, :passed, :runtime].each do |stat|
-                status[stat] = summary[stat] + status[stat]
+            redis.multi do |multi|
+              if value && summary = Sidekiq.load_json(value)[worker]
+                summary = summary.symbolize_keys
+                [:failed, :passed, :runtime].each do |stat|
+                  status[stat] = summary[stat] + status[stat]
+                end
               end
-            end
 
-            multi.set(history, Sidekiq.dump_json({ worker => status}))
-          end
+              multi.set(history, Sidekiq.dump_json(worker => status))
```

Comment:
This still deletes the existing history written for the previous jobs that have been processed, replaced by the last job class processed.